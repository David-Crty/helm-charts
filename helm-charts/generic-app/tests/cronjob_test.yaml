# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json

suite: test cronjob
release:
  name: test
chart:
  version: 0.1.0
templates:
  - cronjob.yaml
tests:
  - it: should have a cronjob
    set:
      cronJobs:
        cron-1:
          command:
            - echo
            - "hello world"
          schedule: "* * * * *"
    asserts:
      - matchSnapshot: {}