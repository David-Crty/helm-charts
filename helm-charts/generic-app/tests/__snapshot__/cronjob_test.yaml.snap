should have a cronjob:
  1: |
    apiVersion: batch/v1
    kind: CronJob
    metadata:
      labels:
        app.kubernetes.io/instance: test
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: generic-app
        helm.sh/chart: generic-app-0.1.0
      name: test-generic-app-cron-1
    spec:
      jobTemplate:
        metadata:
          labels:
            app.kubernetes.io/component: cronjob
            app.kubernetes.io/instance: test
            app.kubernetes.io/name: generic-app
          name: test-generic-app-cron-1
        spec:
          template:
            spec:
              containers:
                - command:
                    - echo
                    - hello world
                  image: hello-world
                  imagePullPolicy: Always
                  name: test-generic-app-cron-1
              restartPolicy: OnFailure
      schedule: '* * * * *'
